apps/meteor/client/components/connectionStatus/ConnectionStatusBar.tsx

```diff
@@ -1,54 +1,14 @@
-import { Icon } from '@rocket.chat/fuselage';
-import { useConnectionStatus, useTranslation } from '@rocket.chat/ui-contexts';
-import type { MouseEventHandler, FC } from 'react';
-import React, { useEffect, useRef, useState } from 'react';
+import { Box, Icon } from '@rocket.chat/fuselage';
+import { useConnectionStatus } from '@rocket.chat/ui-contexts';
+import React, { type MouseEventHandler } from 'react';
+import { useTranslation } from 'react-i18next';
 
-import './ConnectionStatusBar.styles.css';
+import { useReconnectCountdown } from './useReconnectCountdown';
 
-// TODO: frontend chapter day - fix unknown translation keys
-
-const getReconnectCountdown = (retryTime: number): number => {
-	const timeDiff = retryTime - Date.now();
-	return (timeDiff > 0 && Math.round(timeDiff / 1000)) || 0;
-};
-
-const useReconnectCountdown = (
-	retryTime: number | undefined,
-	status: 'connected' | 'connecting' | 'failed' | 'waiting' | 'offline',
-): number => {
-	const reconnectionTimerRef = useRef<ReturnType<typeof setInterval>>();
-	const [reconnectCountdown, setReconnectCountdown] = useState(() => (retryTime ? getReconnectCountdown(retryTime) : 0));
-
-	useEffect(() => {
-		if (status === 'waiting') {
-			if (reconnectionTimerRef.current) {
-				return;
-			}
-
-			reconnectionTimerRef.current = setInterval(() => {
-				retryTime && setReconnectCountdown(getReconnectCountdown(retryTime));
-			}, 500);
-			return;
-		}
-
-		reconnectionTimerRef.current && clearInterval(reconnectionTimerRef.current);
-		reconnectionTimerRef.current = undefined;
-	}, [retryTime, status]);
-
-	useEffect(
-		() => (): void => {
-			reconnectionTimerRef.current && clearInterval(reconnectionTimerRef.current);
-		},
-		[],
-	);
-
-	return reconnectCountdown;
-};
-
-const ConnectionStatusBar: FC = function ConnectionStatusBar() {
+const ConnectionStatusBar = function ConnectionStatusBar() {
 	const { connected, retryTime, status, reconnect } = useConnectionStatus();
 	const reconnectCountdown = useReconnectCountdown(retryTime, status);
-	const t = useTranslation();
+	const { t } = useTranslation();
```

Comment:
I think its not going to work here, in this file

Reply:
Why not?

Reply:
I think there's a verification on the provider and just translations with the proper pattern works with `react-i18next`